### ISSUE TITLE
custom_lint fails with "Failed to decode" on unrelated files.

### ISSUE THREAD

**Issue by felix-barz-brickmakers** [invertase/dart_custom_lint#268]
**Describe the bug**
When trying to run custom_lint on a more complex flutter project, it fails with `Failed to decode .dart_tool/package_config.json` on unrelated files. This happens for:
- Symlinked iOS plugins (when depending on a package that provides a native iOS flutter plugin, i. e. `package_info_plus`
  - `Failed to decode .dart_tool/package_config.json at .../ios/.symlinks/plugins/package_info_plus. Make sure to run pub get first.`
- The synthetic package generated by `flutter_localizations`
  - `Failed to decode .dart_tool/package_config.json at .../.dart_tool/flutter_gen. Make sure to run pub get first.`

The error is the same in all cases (with different paths):
```
PathNotFoundException: Cannot open file, path = '.../.dart_tool/package_config.json' (OS Error: No such file or directory, errno = 2)
#0      _checkForErrorResponse (dart:io/common.dart:55:9)
#1      _File.open.<anonymous closure> (dart:io/file_impl.dart:381:7)
#2      _rootRunUnary (dart:async/zone.dart:1407:47)
#3      _CustomZone.runUnary (dart:async/zone.dart:1308:19)
<asynchronous suspension>
#4      _File.readAsBytes.<anonymous closure> (dart:io/file_impl.dart:562:24)
<asynchronous suspension>
#5      parsePackageConfig (package:custom_lint_core/src/package_utils.dart:161:5)
<asynchronous suspension>
#6      CustomLintProject.parse (package:custom_lint/src/workspace.dart:1025:34)
<asynchronous suspension>
#7      Future.wait.<anonymous closure> (dart:async/future.dart:534:21)
<asynchronous suspension>
#8      CustomLintWorkspace.fromContextRoots (package:custom_lint/src/workspace.dart:516:22)
<asynchronous suspension>
#9      _runServer.<anonymous closure> (package:custom_lint/custom_lint.dart:90:25)
<asynchronous suspension>
#10     _AsyncCompleter.complete (dart:async/future_impl.dart:41:3)
<asynchronous suspension>
#11     _runServer.<anonymous closure> (package:custom_lint/custom_lint.dart:126:19)
<asynchronous suspension>
#12     _runServer (package:custom_lint/custom_lint.dart:86:3)
<asynchronous suspension>
#13     customLint (package:custom_lint/custom_lint.dart:50:5)
<asynchronous suspension>
#14     entrypoint (file:///Users/user/.pub-cache/hosted/pub.dev/custom_lint-0.6.5/bin/custom_lint.dart:68:3)
<asynchronous suspension>
#15     main (file:///Users/user/.pub-cache/hosted/pub.dev/custom_lint-0.6.5/bin/custom_lint.dart:80:5)
<asynchronous suspension>

#0      CustomLintProject.parse.<anonymous closure> (package:custom_lint/src/workspace.dart:1028:7)
#1      _rootRunBinary (dart:async/zone.dart:1423:47)
#2      _CustomZone.runBinary (dart:async/zone.dart:1315:19)
#3      _FutureListener.handleError (dart:async/future_impl.dart:178:22)
#4      Future._propagateToListeners.handleError (dart:async/future_impl.dart:873:47)
#5      Future._propagateToListeners (dart:async/future_impl.dart:894:13)
#6      Future._completeError (dart:async/future_impl.dart:674:5)
<asynchronous suspension>
```

**To Reproduce**
(Example project: [test_project.zip](https://github.com/user-attachments/files/16656145/test_project.zip))
- `flutter create -e test_project`
- `flutter pub add package_info_plus 'flutter_localizations:{"sdk":"flutter"}' dev:custom_lint`
- Setup `custom_lint` and `flutter_localizations`
- `flutter build ios --debug --no-codesign` (To create the `.symlinks` directory)
- Run `custom_lint`

**Expected behavior**
The tool should run without any errors.

**Workaround**
For now, the only workaround seems to be to delete the `ios/.symlinks` and `.dart_tool/flutter_gen` directories prior to executing `dart run custom_lint`.

**Comment by tatsuyukiDunqian** [comment]
I encountered the same issue with custom_lint: 0.6.5, and the problem was only resolved after downgrading to version 0.6.4.

**Comment by lishaduck** [comment]
For me, I ran `flutter clean` and that fixed it, for now. It'll probably break next time I build the app, but it works in CI at least.
